src/audio/module_adapter/module_adapter.c

```diff
@@ -699,7 +735,9 @@ int module_adapter_copy(struct comp_dev *dev)
 out:
 	if (mod->simple_copy) {
 		buffer_release(sink_c);
-		buffer_release(source_c);
+		i = 0;
+		list_for_item(blist, &dev->bsource_list)
+			buffer_release(source_c[i]);
```

Comment:
something's wrong above. You probably meant
```
for (i = 0; i < num_input_buffers; i++)
```

Reply:
there should be symmetry in the acquire/release too.

Reply:
no I didnt, this is balanced with the acquire for all source buffers

Reply:
you see that `i` isn't incremented in the loop above, right? So the loop just repeatedly releases the same buffer. Please, count how many buffers got acquired and use that number for a `for` loop

Reply:
ahh thats a miss. I is incremented in the loop now.